### PR TITLE
Added CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by twolfson**
As discussed in #13, we are adding multiple contributors so we need to document our approval process as well as general contributor guidelines. In this PR:
- Added `CONTRIBUTING.md`

**Notes:**

I diverged from `radiant-player's` CONTRIBUTING.md a bit (which was linked in #13) since there are a few quirks which make our lives easier (e.g. allow branches on repo as long as people clean them up, usage of `/cc` convention).

/cc @MarshallOfSound @jacobwgillespie @chrismou

### CODE REVIEW

**Comment by jacobwgillespie** [comment]
LGTM :+1:

**Comment by MarshallOfSound** [comment]
:+1:

**Comment by twolfson** [comment]
Going to wait for all core contributors to approve this PR before landing it (since unanimity would be great).

**Comment by jacobwgillespie** [comment]
So, that leaves @chrismou, right?  :wink:

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+- Otherwise, please create a new issue
+    - Please restrict issues to a single feature/issue
+    - If the issue is a bug report, then please provide information about your setup as well as steps to reproduce
+- If the issue is time sensitive, then feel free to start work immediately
+    - Please communicate that via the issue (e.g. via a comment on the thread)
+- Otherwise, please wait for a core contributor to approve your work
+    - This is prevent unnecessary rejection when a pull request is opened
+
+## Making changes
+To make sure that there is no duplicate work across contributors, please always [Open an issue][] before beginning work.
```
Comment (chrismou) [inline CONTRIBUTING.md:19]: This link doesn't link anywhere

Comment (twolfson) [inline CONTRIBUTING.md:19]: It's linked below via backreferences. When you use an empty `[]`, Markdown uses the contents of the first brackets so this is effectively:

``` md
[Open an issue][Open an issue]
```

https://github.com/twolfson/google-music.js/pull/17/files#diff-6a3371457528722a734f3c51d9238c13R21

Comment (twolfson) [inline CONTRIBUTING.md:19]: Sorry, the term is "link reference" not backreference =/

http://spec.commonmark.org/0.22/#collapsed-reference-link

Comment (chrismou) [inline CONTRIBUTING.md:19]: Weird. When I clicked it it didn't do anything. May be because I'd already clicked it once before :-/

Comment (twolfson) [inline CONTRIBUTING.md:19]: It jumps up to the previous section so it's prob not too noticable

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+## Submitting changes
+To submit your changes, please open a pull request from your branch to the `google-music.js` `master` branch. If this is for an inter-dependent branch, please still point to the `master` branch. When one PR is landed, then the other should automatically land as well.
+
+While we don't require squashing commits for historical reasons, please avoid unnecessary commits in the final pull request. These can be cleaned up via `squash` in `git rebase -i`.
+
+If there is a particular contributor that should review the pull request, then please add them via a "/cc @username" on the pull request.
+
+## Approval process
+Due to the brittle nature of `google-music.js`, we must be flexible to quickly deploying fixes. Please use the following guidelines when approving/landing changes.
```
Comment (chrismou) [inline CONTRIBUTING.md]: "we must be flexible to allow for quick deployment of fixes" may read a bit better

**Comment by chrismou** [comment]
Aside from nitpicking over some wording and a weird thing I had with an anchor link, all seems good, albeit a bit wordy.  LGTM :+1:

File: CONTRIBUTING.md (hunk 2 above)
Comment (twolfson) [inline CONTRIBUTING.md]: Will update this one, thanks :+1:

**Comment by twolfson** [comment]
Cool, going to fix up the wording comment and land this PR :tada:

**Comment by twolfson** [comment]
This has been landed/deployed in `3.3.2`
